odoo/addons/base/tests/common.py

```diff
@@ -43,40 +102,64 @@ def _enable_currency(cls, currency_code):
         return currency
 
     @classmethod
-    def _use_currency(cls, currency_code):
+    def _use_currency(cls, company, currency_code):
         # Enforce constant currency
         currency = cls._enable_currency(currency_code)
-        if not cls.env.company.currency_id == currency:
-            cls.env.transaction.cache.set(cls.env.company, type(cls.env.company).currency_id, currency.id, dirty=True)
+        if not company.currency_id == currency:
+            cls.env.transaction.cache.set(company, type(company).currency_id, currency.id, dirty=True)
             # this is equivalent to cls.env.company.currency_id = currency but without triggering buisness code checks.
             # The value is added in cache, and the cache value is set as dirty so that that
             # the value will be written to the database on next flush.
             # this was needed because some journal entries may exist when running tests, especially l10n demo data.
 
-class BaseUsersCommon(BaseCommon):
-
     @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-
-        cls.group_portal = cls.env.ref('base.group_portal')
-        cls.group_user = cls.env.ref('base.group_user')
+    def _create_partner(cls, **create_values):
+        return cls.env['res.partner'].create({
+            'name': "Test Partner",
+            'company_id': False,
+            **create_values,
+        })
 
-        cls.user_portal = cls.env['res.users'].create({
-            'name': 'Test Portal User',
-            'login': 'portal_user',
-            'password': 'portal_user',
-            'email': 'portal_user@gladys.portal',
-            'groups_id': [Command.set([cls.group_portal.id])],
+    @classmethod
+    def _create_user(cls, groups, **create_values):
+        return cls.env['res.users'].create({
+            'name': "Test User",
+            'login': 'test_user',
+            'password': 'test_user',
+            'email': 'test_user@test.com',
+            'groups_id': [Command.set(groups.ids)],
+            'company_id': cls.env.company.id,
+            'company_ids': [Command.set(cls.env.company.ids)],
+            **create_values,
         })
 
-        cls.user_internal = cls.env['res.users'].create({
-            'name': 'Test Internal User',
-            'login': 'internal_user',
-            'password': 'internal_user',
-            'email': 'mark.brown23@example.com',
-            'groups_id': [Command.set([cls.group_user.id])],
+    @classmethod
+    def _create_company(cls, **create_values):
+        company = cls.env['res.company'].create({
+            'name': "Test Company",
+            **create_values,
         })
+        cls.env.user.company_ids = [Command.link(company.id)]
+        # cls.env.context['allowed_company_ids'].append(company.id)
+        return company
+
+    @classmethod
+    def _create_internal_user(cls):
```

Comment:
Maybe worth supporting additional create values for this util and the `_create_portal_user` one?
Because right now, they can only be called once, otherwise it'll fail because the same login is used twice.